Xamarin.Forms.Core/ExperimentalFlags.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Xamarin.Forms
+{
+	internal static class ExperimentalFlags
+	{
```

Comment:
I'd prefer if you could keep this logic in some Visual class, and leave the CollectionView where it is too. we do not need to generify this

Reply:
Why? I generalized it to ensure the exception messaging and behavior stays consistent between the features. Also once Shell is rolled in behind a flag it can just use this static method as well and thus maintain consistent flag failing behavior